assets/js/slider.js

```diff
@@ -0,0 +1,283 @@
+// Global universal functions
+function $(elem) {
+  return document.querySelector(elem);
+}
+
+function hasClass(elem, className) {
+  return elem.classList ? elem.classList.contains(className) : false;
+}
+
+function addClass(elem, className) {
+  if (elem.classList) {
+    elem.classList.add(className);
+  } else {
+    elem.className += ' ' + className;
+  }
+}
+
+function removeClass(elem, className) {
+  if (elem.classList) {
+    elem.classList.remove(className);
+  }
+}
+
+const Slider = (function () {
+
+  const slider = function (settings) {
+
+    this.state = {
+      target: settings.target,
+      dotsContainer: settings.dotsContainer,
+      arrowLeft: settings.arrowLeft,
+      arrowRight: settings.arrowRight,
+      transition: settings.transition,
+      swipe: true,
+      autoHeight: true
+    }
+
+    this.init();
+  }
+
+  slider.prototype.buildDots = function () {
+
+    for (let i = 0; i < this.totalSlides; i++) {
+      const dot = document.createElement('div');
+
+      dot.setAttribute('data-slide', i + 1);
+      dot.setAttribute('class', 'slider-dot');
+
+      this.state.dotsContainer.appendChild(dot);
+    }
+
+    this.state.dotsContainer.addEventListener('click', (e) => {
+      if (e.target) {
+        this.currentSlide = e.target.getAttribute('data-slide');
+        this.goToSlide();
+      }
+    }, false);
+  }
+
+  slider.prototype.getCurrentLeft = function () {
+    this.currentLeft = parseInt(this.sliderInner.style.left.split('px')[0]);
+  }
+
+  slider.prototype.goToSlide = function () {
+
+    this.sliderInner.style.transition = 'left ' + this.state.transition.speed / 1000 + 's ' + this.state.transition.easing;
+    this.sliderInner.style.left = -this.currentSlide * this.slideW + 'px';
+
+    addClass(this.state.target, 'isAnimating');
+
+    setTimeout(() => {
+      this.sliderInner.style.transition = '';
+      removeClass(this.state.target, 'isAnimating');
+    }, this.state.transition.speed);
+
+    this.setDot();
```

Comment:
I assume your dot would be set earlier than your slide due to `setTimeout`

Reply:
Done